board/drivers/rtc.h

```diff
@@ -21,9 +22,21 @@ uint16_t from_bcd(uint8_t value){
 }
 
 void rtc_init(void){
+  uint32_t bdcr_opts = 0U;
+  uint32_t bdcr_mask = 0U;
+  if ((hw_type == HW_TYPE_DOS) || (hw_type == HW_TYPE_UNO)) {
```

Comment:
think it makes more sense to keep the has_rtc field, and use it for this check

Reply:
Nope, I need to choose between LSE and LSI clocks. Only Uno and Dos has external 32.768kHz.

Reply:
would still put that in the hw abstraction layer